Export libsession constants

yougotwill opened the pull request:
- added constants wrapper to export user_profile, contact, group and community max name length variables
- user info now truncate names if they are too long
- `setUserInfo` will now return the final name result (in case of truncation)
- added `yarn clean`
- rename `Readme.md` to `README.md`
- include node-api-headers in cmake build process which fixes intellisense issues

yougotwill committed:
added constants wrapper to export MAX_NAME_LENGTH for desktop project

yougotwill committed:
fix: constants now works correctly with correct typings!

updated gitignore file

yougotwill committed:
feat: added group name max lengths

These are the only constants we need for now

yougotwill committed:
feat: we return the user profile name when setUserInfo is called

since we now truncate by default we need to give session-desktop access
to the truncated name result

yougotwill committed:
chore: bump to 0.3.20

Bilb commented on shared.d.ts:
```diff
@@ -64,4 +64,15 @@ declare module 'libsession_util_nodejs' {
 
   export type BaseWrapperActionsCalls = MakeWrapperActionCalls<BaseConfigWrapper>;
+
+  export type ConstantsType = {
+    /** 100 bytes */
+    CONTACT_MAX_NAME_LENGTH: number;
+    /** 100 bytes - for legacy groups and communities */
+    BASE_GROUP_MAX_NAME_LENGTH: number;
+    /** 100 bytes */
+    GROUP_INFO_MAX_NAME_LENGTH: number;
+  }
```
nice to have the jsdoc, but also easy to get out of sync if ever libsession changes it

yougotwill replied:
Yeah not an ideal solution if we can find a way to link or reference the C++ directly then I'm all for it.

Bilb commented:
lgtm

yougotwill committed:
feat: added COMMUNITY_FULL_URL_MAX_LENGTH

cleaned up CONSTANTS object creation

yougotwill committed:
feat: added yarn clean command

yougotwill committed:
fix: rename Readme.md to README.md

yougotwill committed:
fix: include node-api-headers in cmake build process

fixes intellisense issues in vs code

yougotwill committed:
fix: update README release instructions to check libsession_util submodule

reverted version in package.json so we can follow the proper documented process for releases

yougotwill committed:
feat: moved community url max length to cpp in libsession_util

Bilb commented on package.json:
```diff
@@ -9,14 +9,16 @@
     "email": "team@oxen.io"
   },
   "scripts": {
+    "clean": "rimraf .cache && rimraf build",
```
you can just have 
`rimraf .cache build`
that will avoid keeping the build, when cache is empty

yougotwill replied:
Fixed!

yougotwill committed:
fix: clean does directories together
